File: YogaD/app/src/main/java/com/nguyenthanhduy/yoga_d/model/YogaClass.java
package com.nguyenthanhduy.yoga_d.model;

import android.annotation.SuppressLint;

import androidx.annotation.NonNull;

import com.nguyenthanhduy.yoga_d.utils.DateUtil;

import java.io.Serializable;
import java.util.Calendar;

public class YogaClass implements Serializable {
    public String yogaClassID;
    private double price;
    public int dayOfWeek;
    public String timeOfCourse;
    public String typeOfClass;
    public String description;
    public String yogaName;
    public int capacity;
    private int duration;

    public YogaClass() {
        yogaClassID = String.valueOf(Calendar.getInstance().getTime().getTime());
    }

    public YogaClass(String id) {
        this.yogaClassID = id;
    }


    public void setDuration(int duration) {
        this.duration = duration;
    }

    public int getDuration() {
        return this.duration;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public double getPrice() {
        return this.price;
    }

    @SuppressLint("DefaultLocale")
    public String priceString() {
        return String.format("%.2f %s", price, " £");
    }

    @SuppressLint("DefaultLocale")
    public String getDurationString() {
        return String.format("%d %s", this.duration, "mins");
    }

    public String getDayOfWeekString() {
        try {
            return DateUtil.dayOfWeekToString(dayOfWeek);
        } catch (Exception e) {
            return e.getMessage();
        }
    }

    @NonNull
    public String toString() {
        String toString = "Name: " + yogaName + "\n" +
                "Capacity: " + capacity + "\n" +
                "Type of class: " + typeOfClass + "\n" +
                "Day of week: " + getDayOfWeekString() + "\n" +
                "Time of course: " + timeOfCourse + "\n" +
                "Price: " + priceString() + "\n" +
                "Duration: " + getDurationString() + "\n" +
                "Description: " + description;
        return toString;
    }
}

File: yoga_d_client/app/model/YogaClass.cs
namespace yoga_d_client.app.model
{
    public class YogaClass
    {
        public string yogaClassId { get; set; }
        public string yogaName { get; set; }
        public int dayOfWeek { get; set; }
        public string  DayOfWeekString
        {
            get { return dayOfWeekToString(); }
            }
        public string timeOfCourse { get; set; }
        public int capacity { get; set; }
        public int duration { get; set; }
        public double price { get; set; }
        public string typeOfClass { get; set; }
        public string description { get; set; }

        public List<ClassInstance> classInstanceList = new List<ClassInstance>();

        public string dayOfWeekToString()
        {
            switch (dayOfWeek)
            {
                case 1: return "Sunday";
                case 2: return "Monday";
                case 3: return "Tuesday";
                case 4: return "Wednesday";
                case 5: return "Thursday";
                case 6: return "Friday";
                case 7: return "Saturday";
                default: return "Invalid date";
            }
        }

        public string priceToString()
        {
            return price + " £";
        }
        public string durationToString()
        {
            return duration + " mins";
        }
    }
}
